clusterloader2/pkg/framework/framework.go

```diff
@@ -84,16 +88,47 @@ func (f *Framework) CreateAutomanagedNamespaces(namespaceCount int) error {
 	return nil
 }
 
+// ListAutomanagedNamespaces returns all existing automanged namespace names.
+func (f *Framework) ListAutomanagedNamespaces() ([]string, error) {
+	var automanagedNamespacesList []string
+	namespacesList, err := client.ListNamespaces(f.clientSet)
+	if err != nil {
+		return automanagedNamespacesList, err
+	}
+	for _, namespace := range namespacesList {
+		if isAutomanagedNamespace(namespace) {
+			automanagedNamespacesList = append(automanagedNamespacesList, namespace)
+		}
+	}
+	return automanagedNamespacesList, nil
+}
+
 // DeleteAutomanagedNamespaces deletes all automanged namespaces.
-func (f *Framework) DeleteAutomanagedNamespaces() error {
+func (f *Framework) DeleteAutomanagedNamespaces() []error {
+	var wg wait.Group
+	var lock sync.Mutex
+	var errList []error
```

Comment:
It seems like it's finally time to implement this errList class (maybe in another PR) instead of adding all this locking code everywhere.

Reply:
Good idea. I will make separate PR for this.

Reply:
Sure, but please remember to do it (or add TODO somewhere and fix).

Reply:
Done.